ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/util/SimpleOffsetSerializer.java

```diff
@@ -136,40 +137,44 @@ public static void setConstants() {
   }
 
   public static Bytes serialize(SimpleOffsetSerializable value) {
-    List<UInt64> variable_offsets = new ArrayList<>();
-    List<Bytes> interleaved_values = new ArrayList<>();
-    UInt64 fixedLengthSum = UInt64.ZERO;
-    UInt64 varLengthSum = UInt64.ZERO;
-
-    for (Bytes fixedPart : value.get_fixed_parts()) {
-      UInt64 fixedPartSize = UInt64.valueOf(fixedPart.size());
-      if (fixedPartSize.equals(UInt64.ZERO)) {
-        fixedPartSize = UInt64.valueOf(4L);
+    if (value instanceof ViewRead) {
+      return ((ViewRead) value).sszSerialize();
+    } else {
```

Comment:
nit: I'd just omit the else here since the if block returns anyway. Saves indenting the rest of the method.